static/scripts.js

```diff
@@ -39,40 +41,93 @@ var mud = {}; // Namespace
 
 	mud.onEntryKeyPress = function (oCtl, oEvent) {
 		if (mud.isEnterKeyPress(oEvent)) {
-			// Capture the current text as a command
-			var sEntry = oCtl.value;
+			mud.submitCommand(oCtl);
+		}
+		else if(mud.isUpKeyPress(oEvent)) {
+			mud.recallNextHistoricalCommand(oCtl);
+		}
+		else if(mud.isDownKeyPress(oEvent)) {
+			mud.recallPrevHistoricalCommand(oCtl);
+		}
+	};
 
-			// Reset the text entry for the next command
-			oCtl.value = '';
+	mud.submitCommand = function (oCtl) {
+		// Capture the current text as a command
+		var sEntry = oCtl.value;
 
-			if (mud.user === null) {
-				// Set the username first if we still need one
-				if (sEntry.length > 0) {
-					mud.user = sEntry;
-					mud.writeOutput('Type \'help\' for a list of commands');
-				}
-			}
-			else {
-				// Process the entry
-				mud.idleTicks = 0;
-				mud.processEntry(sEntry);
+		// Reset the text entry for the next command
+		oCtl.value = '';
+
+		if (mud.user === null) {
+			// Set the username first if we still need one
+			if (sEntry.length > 0) {
+				mud.user = sEntry;
+				mud.writeOutput('Type \'help\' for a list of commands');
 			}
 		}
+		else {
+			// Add the command to the beginning of the command history
+			mud.previousCommands.unshift(sEntry);
+			//  Reset our place in the previous commands list to the beginning
+			mud.previousCommandsIndex = -1;
+
+			// Process the entry
+			mud.idleTicks = 0;
+			mud.processEntry(sEntry);
+		}
 	};
 
-	mud.isEnterKeyPress = function (oEvent) {
-		var keynum;
+	mud.recallNextHistoricalCommand = function (oCtl) {
+		//  Ensure that we aren't at the end of our historical command list
+		if(mud.previousCommandsIndex + 1 <= mud.previousCommands.length - 1)
+		{
+			mud.previousCommandsIndex++;
+			oCtl.value = mud.previousCommands[mud.previousCommandsIndex];
+		}
+	};
 
-		if (window.event) { // IE8 and earlier
-			keynum = oEvent.keyCode;
-		} else if (oEvent.which) { // IE9/Firefox/Chrome/Opera/Safari
-			keynum = oEvent.which;
+	mud.recallPrevHistoricalCommand = function (oCtl) {
+		//  Ensure that we aren't going past the beginning of our historical command list
+		if(mud.previousCommandsIndex - 1 >= 0)
+		{
+			mud.previousCommandsIndex--;
+			oCtl.value = mud.previousCommands[mud.previousCommandsIndex];
 		}
+	};
+
+	mud.isEnterKeyPress = function (oEvent) {
+		var keynum = mud.captureKeyNum(oEvent);
 
 		// Detect ENTER key
 		return ('\n' === String.fromCharCode(keynum) || '\r' === String.fromCharCode(keynum));
 	};
 
+	mud.isUpKeyPress = function (oEvent) {
+		var keynum = mud.captureKeyNum(oEvent);
+
+		// Detect UP key
+		return (38 == keynum);
+	};
+
+	mud.isDownKeyPress = function (oEvent) {
+		var keynum = mud.captureKeyNum(oEvent);
+
+		// Detect DOWN key
+		return (40 == keynum);
+	};
+
+	mud.captureKeyNum = function (oEvent) {
```

Comment:
Encapsulated this into it's own function rather than having it repeat in three places.